node/src/types/block.rs

```diff
@@ -774,7 +774,12 @@ impl Block {
             parent_hash,
             state_root_hash,
             body_hash,
-            deploy_hashes: finalized_block.proto_block.wasm_deploys,
+            deploy_hashes: finalized_block
+                .proto_block
+                .wasm_deploys
+                .into_iter()
+                .chain(finalized_block.proto_block.transfers.into_iter())
+                .collect(),
```

Comment:
Actually, would it make sense to make this a `deploys` method of `FinalizedBlock`, too, just like `Block::deploys`?

Reply:
`BlockLike::deploys()` combines the two lists. I missed this place during my refactoring.

Reply:
I guess it would. Would you make it return an iterator, or a vector?

Reply:
It returns a `Vec<&DeployHash>`, though, so I'd have to iterate and clone every hash. It would probably make more sense if the method returned an iterator, so that the caller could collect into a type of their choosing, wouldn't it?

Reply:
Ah, except this is a trait method, so the return type would have to be a boxed iterator... But maybe that's still somewhat better?

Reply:
Actually, it looks like material for a separate task :grimacing: I could merge this as is, or I could call `deploys().into_iter().cloned().collect()`, what do you guys think?

Reply:
I think it's fine to merge it as-is and create a tech-debt task.

Reply:
I created https://casperlabs.atlassian.net/browse/NDRS-689, then :+1: